src/pages/ThorChainLP/components/ReusableLpStatus/ReusableLpStatus.tsx

```diff
@@ -0,0 +1,129 @@
+import {
+  Button,
+  CardBody,
+  CardFooter,
+  Center,
+  CircularProgress,
+  Flex,
+  Heading,
+  HStack,
+} from '@chakra-ui/react'
+import type { AssetId } from '@shapeshiftoss/caip'
+import { thorchainAssetId } from '@shapeshiftoss/caip'
+import type { Asset } from '@shapeshiftoss/types'
+import type { PropsWithChildren } from 'react'
+import { useMemo } from 'react'
+import { FaCheck } from 'react-icons/fa'
+import { useTranslate } from 'react-polyglot'
+import { rune } from 'test/mocks/assets'
+import { Amount } from 'components/Amount/Amount'
+import { SlideTransition } from 'components/SlideTransition'
+import { RawText } from 'components/Text'
+import { AsymSide, type ConfirmedQuote } from 'lib/utils/thorchain/lp/types'
+import type { ParsedPool } from 'pages/ThorChainLP/queries/hooks/usePools'
+import { selectAssetById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+type ReusableLpStatusProps = {
+  handleBack: () => void
+  pool?: ParsedPool
+  baseAssetId: AssetId
+  isComplete?: boolean
+  confirmedQuote: ConfirmedQuote
+} & PropsWithChildren
+
+export const ReusableLpStatus: React.FC<ReusableLpStatusProps> = ({
+  pool,
+  baseAssetId,
+  isComplete,
+  confirmedQuote,
+  handleBack,
+  children,
+}) => {
+  const translate = useTranslate()
+
+  const asset = useAppSelector(state => selectAssetById(state, pool?.assetId ?? ''))
+  const baseAsset = useAppSelector(state => selectAssetById(state, baseAssetId))
+
+  const assets: Asset[] = useMemo(() => {
+    if (!(pool && asset && baseAsset)) return []
+
+    if (pool.asymSide === null) return [asset, baseAsset]
+    if (pool.asymSide === AsymSide.Rune) return [baseAsset]
+    if (pool.asymSide === AsymSide.Asset) return [asset]
+
+    throw new Error('Invalid asym side')
```

Comment:
if you use a switch statement with `assertUnreachable(asymSide)` you can enforce exhaustiveness at compile time with a catch-all error as worst case

Reply:
Here too @reallybeard